examples/mnist/eth_mnist.py

```diff
@@ -80,11 +83,30 @@
 network.add_monitor(inh_voltage_monitor, name="inh_voltage")
 
 # Load MNIST data.
-images, labels = MNIST(
-    path=os.path.join("..", "..", "data", "MNIST"), download=True
-).get_train()
-images = images.view(-1, 784)
-images *= intensity
+# images, labels
+dataset = MNIST(
+    None,
+    None,
+    root=os.path.join("..", "..", "data", "MNIST"),
+    download=True,
+    transform=transforms.Compose(
+        [
+            transforms.ToTensor(),
+            transforms.Lambda(lambda x: x * intensity),
+            transforms.Lambda(lambda x: x.view(784)),
+        ]
+    ),
+)
+# .get_train()
+
+images = []
+labels = []
+
+for i in range(len(dataset)):
+    sample = dataset[i]
+    images.append(sample["image"])
+    labels.append(sample["label"])
+
```

Comment:
I'm not sure how I feel about this refactor. It would be difficult for somebody to take this and apply it to a larger dataset because of the extraction of this data ahead of time. I'll leave the final call on this to somebody else though.

Reply:
Yeah, this doesn't make sense. @coopersigrist You should pass a `PoissonEncoder` to the `MNIST` constructor, and then just iterate over it during training.

Reply:
Agreed

Reply:
Agreed! I'm gonna start redoing this now. I'll close this request and make the changes